x-pack/plugins/uptime_monitoring/public/components/queries/monitor_charts/monitor_charts.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  // @ts-ignore missing typings
+  EuiAreaSeries,
+  EuiFlexGroup,
+  EuiFlexItem,
+  // @ts-ignore missing typings
+  EuiLineSeries,
+  EuiPanel,
+  // @ts-ignore missing typings
+  EuiSeriesChart,
+  // @ts-ignore missing typings
+  EuiSeriesChartUtils,
+  // @ts-ignore missing typings
+  EuiSpacer,
+  // @ts-ignore missing typings
+  EuiTitle,
+} from '@elastic/eui';
+import React, { Fragment } from 'react';
+import { Query } from 'react-apollo';
+import { createGetMonitorChartsQuery } from './get_monitor_charts';
+
+interface MonitorChartsProps {
+  dateRangeStart: number;
+  dateRangeEnd: number;
+  monitorId: string;
+  autorefreshInterval: number;
+  autorefreshEnabled: boolean;
+}
+
+export const MonitorCharts = ({
+  dateRangeStart,
+  dateRangeEnd,
+  monitorId,
+  autorefreshEnabled,
+  autorefreshInterval,
+}: MonitorChartsProps) => (
+  <Query
+    pollInterval={autorefreshEnabled ? autorefreshInterval : undefined}
+    query={createGetMonitorChartsQuery}
+    variables={{ dateRangeStart, dateRangeEnd, monitorId }}
+  >
+    {({ loading, error, data }) => {
+      if (loading) {
+        return 'Loading...';
+      }
+      if (error) {
+        return `Error ${error.message}`;
+      }
+      const { monitorChartsData } = data;
+      // console.log(monitorChartsData);
+      const rttWriteRequestSeries: any[] = [];
+      const rttValidateSeries: any[] = [];
+      const rttContentSeries: any[] = [];
+      const rttResponseSeries: any[] = [];
+      const rttTcpSeries: any[] = [];
+      const avgDurationSeries: any[] = [];
+      const areaRttSeries: any[] = [];
+      const downSeries: any[] = [];
+      const upSeries: any[] = [];
+      const checksSeries: any[] = [];
+      monitorChartsData.forEach(
+        ({
```

Comment:
Might just be me and my way of thinking... but I am thinking a reduce to an object where the series are keys might be easier to read? Also it would then give you free typing vs all the any's here

Reply:
I think that this needs to be moved to the server as well - the GQL endpoint should be returning the data that UI needs pre-formatted. I'll make a note of that.

https://github.com/elastic/kibana/pull/27279/commits/0c3b53796f185eab80dd938598c0ed66ac6789ae